src/presets/indexer.ts

```diff
@@ -0,0 +1,54 @@
+import fs from 'node:fs/promises';
+
+import type { IndexInput, IndexerOptions } from '@storybook/types';
+import { preprocess } from 'svelte/compiler';
+
+import { loadSvelteConfig } from './svelte/config-loader.js';
+import type { StoriesFileMeta } from '../parser/types.js';
+import { extractStories } from '../parser/extract-stories.js';
+
+export async function readStories(fileName: string): Promise<StoriesFileMeta> {
+  let code = (await fs.readFile(fileName, { encoding: 'utf8' })).toString();
+
+  const svelteOptions = await loadSvelteConfig();
+
+  if (svelteOptions && svelteOptions.preprocess) {
+    code = (
+      await preprocess(code, svelteOptions.preprocess, {
+        filename: fileName,
+      })
+    ).code;
+  }
+
+  return extractStories(code);
+}
+
+// FIXME: Still want to support it?
```

Comment:
given we're releasing this as a major, this is a good time to remove this support.